prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -33,23 +31,28 @@ class KubernetesProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments: Namespace):
+    def __init__(self, kubeconfig_file, context, namespace, config_file, fixer_config):
         """
         Initializes the KubernetesProvider instance.
         Args:
-            arguments (dict): A dictionary containing configuration arguments.
+            kubeconfig_file (str): Path to the kubeconfig file.
+            context (str): Context name.
+            namespace (list): List of namespaces.
+            config_file (str): Path to the configuration file.
+            fixer_config (str): Path to the fixer configuration file
         """
+
         logger.info("Instantiating Kubernetes Provider ...")
-        self._session = self.setup_session(arguments.kubeconfig_file, arguments.context)
-        if not arguments.namespace:
+        self._session = self.setup_session(kubeconfig_file, context)
+        if not namespace:
             logger.info("Retrieving all namespaces ...")
             self._namespaces = self.get_all_namespaces()
         else:
-            self._namespaces = arguments.namespace
+            self._namespaces = namespace
 
         if not self._session.api_client:
             logger.critical("Failed to set up a Kubernetes session.")
-            sys.exit(1)
+            raise SystemExit
```

Comment:
Add a TODO here to replace it with our custom exceptions once they are ready.

Reply:
Done